rfc/composite_table_availability.pseudocode

```diff
@@ -0,0 +1,123 @@
+
+PhysicalTableSchema:
+	getFieldName(Column c)
+
+DataSource:
+	String getName()
+	Map<FieldName, Set<Segment>> raw segment data
+	Map<FieldName, List<Interval>> squashedFromSegments  // optional optimization
```

Comment:
Talking to @garyluoex today another reason to shun SimplifiedIntervalList is that it consciously doesn't support all list operations.  Using primarily immutable lists may be the smarter choice.

ps - unmodifiableCollections don't observe the collections equality contract, instead use the Object equality contract.  fun fact.